CONTRIBUTING.rst

```diff
@@ -0,0 +1,247 @@
+  `standard <http://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt#docstring-standard>`__
+  as NumPy and SciPy.
+* No changes are ever committed without review.  Ask on the
+  `mailing list <http://groups.google.com/group/mousestyles>`_ if
+  you get no response to your pull request.
+  **Never merge your own pull request.**
+
+Stylistic Guidelines
+--------------------
+
+* Set up your editor to remove trailing whitespace.  Follow `PEP08
+  <www.python.org/dev/peps/pep-0008/>`__.  Check code with pyflakes / flake8.
+
+* Use numpy data types instead of strings (``np.uint8`` instead of
+  ``"uint8"``).
+
+* Use the following import conventions::
+
+   import numpy as np
+   import scipy as sp
+   import matplotlib as mpl
+   import matplotlib.pyplot as plt
+
+   cimport numpy as cnp # in Cython code
+
+Commit message codes
+---------------------
+
+Please prefix all commit summaries with one (or more) of the following labels.
+This should help others to easily classify the commits into meaningful
+categories:
+
+* *BUG* : bug fix
+* *RFT* : refactoring
+* *ENH* : new feature or extended functionality
+* *BKW* : addresses backward-compatibility
+* *OPT* : optimization
+* *BRK* : breaks something and/or tests fail
+* *DOC*: for all kinds of documentation related commits
+* *TST* : for adding or changing tests
+* *DAT* : for adding or changing data files
+* *STY* : PEP8 conformance, whitespace changes etc that do not affect
+  function.
+
+So your commit message might look something like this::
+
+    TST: relax test threshold slightly
+
+    Attempted fix for failure on windows test run when arrays are in fact
+    very close (within 6 dp).
+
+Keeping up a habit of doing this is useful because it makes it much easier to
+see at a glance which changes are likely to be important when you are looking
+for sources of bugs, fixes, large refactorings or new features.
+
+Pull request codes
+------------------
+
+When you submit a pull request to github, github will ask you for a summary.  If
+your code is not ready to merge, but you want to get feedback, please consider
+using ``WIP - experimental optimization`` or similar for the title of your pull
+request. That way we will all know that it's not yet ready to merge and that
+you may be interested in more fundamental comments about design.
+
+When you think the pull request is ready to merge, change the title (using the
+*Edit* button) to something like ``MRG - optimization``.
+
+Test coverage
+-------------
+
+Tests for a module should ideally cover all code in that module,
+i.e., statement coverage should be at 100%.
+
+To measure the test coverage, install
+`coverage.py <http://nedbatchelder.com/code/coverage/>`__
+(e.g., using ``pip install coverage``) and then run::
+
+  $ make coverage
+
+This will print a report with one line for each file in `mousestyles`,
+detailing the test coverage::
+
+  Name                 Stmts   Miss Branch BrMiss  Cover   Missing

```

Comment:
These tests are from *permute*. We can either remove or add the mousestyles output.